service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java

```diff
@@ -675,24 +677,135 @@ private boolean getManualBlockStatus(String userId) {
+            okapiHoldings.forEach(holding -> {
+                String hrid = holding.at("/hrid").asText();
+                String fallbackLocationCode = locationMap.get(holding.at("/permanentLocationId").asText());
+
+                //get items for holding from okapi
+                Map<String, Map<String,String>> okapiItems = getOkapiItems(holding.at("/id").asText(), locationMap);
+
+                //combine marc based holding data and direct okapi data
+                HoldingsRecord recordValues = marcHoldings.get(0);
+                HoldingsRecord currentHolding = HoldingsRecord.builder()
+                                .recordId(recordValues.getRecordId())
+                                .marcRecordLeader(recordValues.getMarcRecordLeader())
+                                .mfhd(hrid)
+                                .issn(recordValues.getIssn())
+                                .isbn(recordValues.getIsbn())
+                                .title(recordValues.getTitle())
+                                .author(recordValues.getAuthor())
+                                .publisher(recordValues.getPublisher())
+                                .place(recordValues.getPlace())
+                                .year(recordValues.getYear())
+                                .genre(recordValues.getGenre())
+                                .fallbackLocationCode(fallbackLocationCode)
+                                .edition(recordValues.getEdition())
+                                .oclc(recordValues.getOclc())
+                                .recordId(recordValues.getRecordId())
+                                .callNumber(recordValues.getCallNumber())
+                                .largeVolume(recordValues.isLargeVolume())
+                                .catalogItems(okapiItems.size() > 0 ? okapiItems:recordValues.getCatalogItems())
+                                .build();
+
+                finalHoldings.add(currentHolding);
+
+                logger.debug("Record ID: {}", currentHolding.getRecordId());
+                logger.debug("Marc record leader: {}", currentHolding.getMarcRecordLeader());
+                logger.debug("MFHD: {}", currentHolding.getMfhd());
+                logger.debug("ISBN: {}", currentHolding.getIsbn());
+                logger.debug("Fallback location: {}", currentHolding.getFallbackLocationCode());
+                logger.debug("Call number: {}", currentHolding.getCallNumber());
+                logger.debug("Valid large volume: {}", currentHolding.isLargeVolume());
+            });
+
         } catch (DOMException | IOException | ParserConfigurationException | SAXException e) {
             // TODO: consider throwing all of these so that caller can handle more appropriately.
             e.printStackTrace();
         }
 
-        return holdings;
+        return finalHoldings;
+    }
+
+    private Map<String,String> getOkapiLocations() {
```

Comment:
I don’t think we will need to get all locations. We can use the caching get locations method. 

https://github.com/TAMULib/CatalogService/blob/50648e520d372084c5728d9e66948902298d840a/service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java#L1193

Reply:
Ok, thanks. Updated the code to use getLocation().

I either missed that method or it wasn't in the branch I started from.